xarray/core/variable.py

```diff
@@ -207,6 +210,8 @@ def _as_array_or_item(data):
 
     TODO: remove this (replace with np.asarray) once these issues are fixed
     """
+    if hasattr(data, '__array_function__'):
```

Comment:
This function is called inside `Variable.values` for coercing into NumPy arrays.

Instead, I would suggest changing the conditional check inside the `data` property, so it return `self._data` directly for types that aren't just dask arrays:
https://github.com/pydata/xarray/blob/82fd8320b1be7a434effd247baf07004b66b802f/xarray/core/variable.py#L289-L294